test/bigtable/family.js

```diff
@@ -0,0 +1,318 @@
+    it('should format the column family name', function() {
+      var formatted = Family.formatName_(TABLE.id, FAMILY_NAME);
+
+      assert.strictEqual(formatted, FAMILY_ID);
+    });
+
+    it('should not re-format the name', function() {
+      var formatted = Family.formatName_(TABLE.id, FAMILY_ID);
+
+      assert.strictEqual(formatted, FAMILY_ID);
+    });
+  });
+
+  describe('formatRule_', function() {
+    it('should capture the max age option', function() {
+      var originalRule = {
+        age: 10
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        maxAge: originalRule.age
+      });
+    });
+
+    it('should capture the max number of versions option', function() {
+      var originalRule = {
+        versions: 10
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        maxNumVersions: originalRule.versions
+      });
+    });
+
+    it('should create a union rule', function() {
+      var originalRule = {
+        age: 10,
+        union: true
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        union: {
+          rules: [{
+            maxAge: originalRule.age
+          }]
+        }
+      });
+    });
+
+    it('should create an intersecting rule', function() {
+      var originalRule = {
+        versions: 2,
+        intersection: true
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        intersection: {
+          rules: [{
+            maxNumVersions: originalRule.versions
+          }]
+        }
+      });
+    });
+  });
+
+  describe('getMetadata', function() {
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+
+      family.parent.getFamilies = function(callback) {
+        callback(err);
+      };
+
+      family.getMetadata(function(err_) {
+        assert.strictEqual(err, err_);
+        done();
+      });
+    });
+
+    it('should update the metadata', function(done) {
+      var FAMILY = new Family(TABLE, FAMILY_NAME);
+      var response = {
+        families: {}
+      };
+
+      FAMILY.metadata = {
+        a: 'a',
+        b: 'b'
+      };
+
+      family.parent.getFamilies = function(callback) {
+        callback(null, [FAMILY], response);
+      };
+
+      family.getMetadata(function(err, metadata, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(FAMILY.metadata, metadata);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+
+    it('should throw a custom error', function(done) {
+      var response = {};
+
+      family.parent.getFamilies = function(callback) {
+        callback(null, [], response);
+      };
+
+      family.getMetadata(function(err, metadata, apiResponse) {
+        assert.strictEqual(err.code, 404);
+        assert.strictEqual(err.message,
```

Comment:
Can you export `Family Error` and check how it behaves separately? Then check that this callback received a `FamilyError`.